### ISSUE TITLE
Feature Request: can we use asynchronous transporters? 

### ISSUE THREAD

**Issue by jeanhdev** [fullstack-build/tslog#229]
I'm using `Axiom` and would love to know if it's possible to use `tslog` with an asynchronous Transporter. 

In `winston` - it's not possible somehow, very weird.

**Comment by terehov** [comment]
I am not familiar with Axiom and don't quite understand the use case. Can you please provide some more context and ideally an example how you would like to use it? 
But I believe from what I understand it should be possible already.

**Comment by CanRau** [comment]
I imagine there's no async as you don't want your system to wait for logging so instead you'd use streaming

here's a tslog example https://tslog.js.org/#/?id=storing-logs-in-a-file-system-with-rotating-files and axioms API also accepts [streaming data](https://axiom.co/docs/send-data/ingest#ingest-api), here's an example on how to [ingest a raw buffer into axiom](https://github.com/axiomhq/axiom-js/blob/edd7495c45923de4826b6db1e9eca03124aa079d/examples/js/src/ingest-file.ts#L8), [here](https://github.com/axiomhq/axiom-js/blob/edd7495c45923de4826b6db1e9eca03124aa079d/packages/js/src/client.ts#L37) you can see that you should be able to pass in a readable stream to `ingestRaw` and maybe the [winston implementation](https://github.com/axiomhq/axiom-js/blob/edd7495c45923de4826b6db1e9eca03124aa079d/packages/winston/src/index.ts#L2) for axiom can help

**Comment by terehov** [comment]
Yes — supported in the upcoming **v5**. Attached transports may be async: `write` can return a `Promise`, `logger.flush()` awaits in-flight writes, and there are batched built-in transports (e.g. `tslog/transports/http`) plus ready-made integration guides. Perfect fit for shipping to Axiom. Closing as addressed in v5.
